src/lib/table/tableData.ts

```diff
@@ -35,67 +35,88 @@ export const dateFormatter = new Intl.DateTimeFormat(undefined, {
+ * @param {ChartSeries[] | undefined} chartSeries - The array with the chart configuration per series.
+ * @param {Record<string, Series>} seriesById - An object with time series with data resource IDs as keys.
+ * @param {string[]} seriesIds - An array of unique series IDs.
  * @returns {TableData[]} - An array of records containing table data.
  */
 export function createTableData(
-  chartSeriesArray: ChartSeries[] | undefined,
-  seriesRecord: Record<string, Series>,
+  chartSeries: ChartSeries[],
+  seriesById: Record<string, Series>,
   seriesIds: string[],
 ): TableData[] {
-  if (chartSeriesArray === undefined) return []
-  const dateTimes = createDateTimes(
-    chartSeriesArray.map((s) => s.dataResources[0]),
-    seriesRecord,
+  // There may be duplicate IDs in the chartSeries array (e.g. for series with
+  // lines and markers for the same data), but the seriesIds array should only
+  // contain unique IDs. Hence, we do a map with a lookup over these IDs rather
+  // than a filter.
+  const usedChartSeries = seriesIds.map((id) =>
+    chartSeries.find((series) => series.id === id),
   )
 
-  const chartSeries = uniqWith(
-    chartSeriesArray.filter((s) => seriesIds.includes(s.id)),
-    (a, b) => {
-      return a.id === b.id
-    },
-  )
-  const pointers = Array(seriesIds.length).fill(0)
-
-  const data = dateTimes.map((date: Date) => {
-    const result: TableData = { date }
-    for (const j in chartSeries) {
-      const s = chartSeries[j]
-      const series = seriesRecord[s.dataResources[0]]
-      let eventResult: Partial<TableSeriesData> = {}
-      if (series && series.data) {
-        const event = series.data[pointers[j]] as TimeSeriesData
-        if (event && date.getTime() === event.x.getTime()) {
-          eventResult = {
-            tooltip: event.flag !== undefined || event.comment !== undefined,
-            ...event,
-          }
-          if (event.flag !== undefined && store.flags !== undefined) {
-            const flag = store.flags.find((f) => f.flag === event.flag)
-            if (flag !== undefined) {
-              eventResult.flagOrigin = flag.source
-              eventResult.flagQuality = flag.quality
-              eventResult.flagEdit = getFlagEdit(
-                event.flag,
-                event.flagSource,
-                flag?.quality,
-              )
-              eventResult.flagColor = getFlagColor(
-                eventResult.flagEdit,
-                flag.quality,
-              )
-            }
-          }
-          pointers[j]++
+  // For each chart series, add its data to a map with the date as key. This
+  // allows us to easily merge the data from different series into a single
+  // record.
+  const data = new Map<Date, TableData>()
+  usedChartSeries.forEach((chartSeries) => {
+    if (!chartSeries) return
+
+    const dataResourceId = chartSeries.dataResources[0]
+    const seriesData = seriesById[dataResourceId].data
+    if (!seriesData) return
```

Comment:
Removed the last commit "fixing" watchers, turns out it is not quite so simple, since some of the properties are actually Proxy objects 😞  for these, reactivity does not work on getters